tests/ha/multi_pod_failure/test_multi_server_pod_failure.py

```diff
@@ -210,195 +182,109 @@ def teardown_method(self):
-
-        LOGGER.info("Step 4: Check cluster status")
-        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
-        assert_utils.assert_false(resp[0], resp)
-        LOGGER.info("Step 4: Cluster has some failures due to %s server pods has gone down.",
-                    self.kvalue)
-
-        LOGGER.info("Step 5: Check services status that were running on server pods"
-                    " which are deleted.")
-        counter = 0
-        for pod_name in self.pod_name_list:
-            hostname = self.pod_dict.get(pod_name)[0]
-            resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=[pod_name], fail=True,
-                                                               hostname=hostname)
-            LOGGER.debug("Services status on %s : %s", pod_name, resp)
-            if not resp[0]:
-                counter += 1
-            server_data_pod_list.remove(pod_name)
-        assert_utils.assert_equal(counter, 0, "Services on some server pods not stopped.")
-        LOGGER.info("Step 5: Services of deleted server pods are in offline state")
-
-        LOGGER.info("Step 6: Check services status on remaining pods %s", server_data_pod_list)
-        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=server_data_pod_list,
-                                                           fail=False)
-        LOGGER.debug("Response: %s", resp)
-        assert_utils.assert_true(resp[0], resp)
-        LOGGER.info("Step 6: Services of remaining pods %s are in "
-                    "online state", server_data_pod_list)
-
-        LOGGER.info("Step 7: Perform READs and verify DI on the written data")
+        self.restore_method = const.RESTORE_DEPLOYMENT_K8S
+        self.restore_pod = True
+        assert_utils.assert_true(resp[0], "Cluster/Services status is not as expected")
+        LOGGER.info("Step 2: Successfully shutdown random %s (K) server pod %s. Verified"
+                    " cluster and services states are as expected & remaining pods status "
+                    "is online.", self.kvalue, self.pod_name_list)
+        LOGGER.info("Step 3: Perform READs and verify DI on the written data")
         resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
                                                     log_prefix=self.test_prefix, skipwrite=True,
-                                                    skipcleanup=True, nsamples=2, nclients=2)
+                                                    skipcleanup=True, nsamples=2, nclients=2,
+                                                    setup_s3bench=False)
         assert_utils.assert_true(resp[0], resp[1])
-        LOGGER.info("Step 7: Performed READs and verified DI on the written data")
-
-        LOGGER.info("Completed: Test to verify degraded READs after all K server pods "
-                    "down - unsafe shutdown.")
+        LOGGER.info("Step 3: Performed READs and verified DI on the written data")
```

Comment:
We decided to add/Keep check for IAM user creation creation and IOs creation in this type of test right @sonalk0209 .

can we add something like,
```suggestion
               if CMN_CFG["dtm0_disabled"]:
            LOGGER.info("STEP 4: Create IAM user and perform WRITEs-READs-Verify-DELETEs with "
                        "variable object sizes on degraded cluster")
            users = self.mgnt_ops.create_account_users(nusers=1)
            self.test_prefix = 'test-40567-1'
            self.s3_clean.update(users)
            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
                                                        log_prefix=self.test_prefix,
                                                        nsamples=2, nclients=2, setup_s3bench=False)
        else:
            LOGGER.info("STEP 4: Perform WRITEs-READs-Verify with variable object sizes on "
                        "degraded cluster")
            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
                                                        log_prefix=self.test_prefix,
                                                        skipcleanup=True, nsamples=2, nclients=2,
                                                        setup_s3bench=False)
        assert_utils.assert_true(resp[0], resp[1])
        LOGGER.info("Step 4: Performed IOs with variable sizes objects.")
```

Reply:
For Reads only, I don't think we need this.